oem/firstvoices/ios/FirstVoices/LexicalModelRepository.swift

```diff
@@ -0,0 +1,83 @@
+/*
+ * LexicalModelRepository.swift
+ * FirstVoices app
+ *
+ * License: MIT
+ *
+ * Copyright © 2022 FirstVoices.
+ *
+ * Created by Shawn Schantz on 2022-01-27.
+ *
+ * Class to call the Keyman API to query about lexical models for a given language.
+ *
+ */
+
+import UIKit
+import KeymanEngine
+
+class LexicalModelRepository {
+
+  private static let keymanLexicalModelApiUrl = "https://api.keyman.com/model?q=bcp47:"
+
+  static let shared: LexicalModelRepository  = {
+    let instance = LexicalModelRepository()
+    return instance
+  }()
+    
+  private init() {}
+
+  func getAvailableLexicalModels(languageTag: String) -> [FVLexicalModel] {
+    var modelArray: [FVLexicalModel] = []
+    let keymanApiUrl: URL = URL.init(string: "\(LexicalModelRepository.keymanLexicalModelApiUrl)\(languageTag)")!
+    // UIApplication.shared.openURL(keymanApiUrl)
+    
+    do {
+      let lexicalModelData = try Data(contentsOf: keymanApiUrl, options: NSData.ReadingOptions())
+
+      do {
+          // make sure this JSON is in the format we expect
+          if let jsonArray = try JSONSerialization.jsonObject(with: lexicalModelData, options: []) as? [[String : Any]] {
+              // try to read array
+            if !jsonArray.isEmpty {
+              let jsonModelMap = jsonArray.first
+              let name = jsonModelMap!["description"] as! String
```

Comment:
Shouldn't name be?
```suggestion
              let name = jsonModelMap!["name"] as! String
Write Preview
```

Reply:
Good question. I used the description field because it had the more succinct name:
        "description": "SEN\u0106O\u0166EN (Saanich Dialect)",
vs
        "name": "SEN\u0106O\u0166EN (Saanich Dialect) Lexical Model"

Not sure what I can expect from each field of the API, but this text goes in the switch name, so I need something brief. Is the lexical model definition incorrect?

Reply:
You should be using `name`. `description` is a long form which may be several lines long. It just so happens that the SNECOTEN model has a overly short description:

https://github.com/keymanapp/lexical-models/blob/fb306c0768caaf6f25cbb7abf9d7e4eb3ca74ef3/release/nrc/nrc.str.sencoten/nrc.str.sencoten.model_info#L3

vs name:

https://github.com/keymanapp/lexical-models/blob/fb306c0768caaf6f25cbb7abf9d7e4eb3ca74ef3/release/nrc/nrc.str.sencoten/source/nrc.str.sencoten.model.kps#L10

See for example https://api.keyman.com/model?q=bcp47:cmo-Khmr:

```json
[
    {
        "license": "mit",
        "languages": [
            "cmo-khmr"
        ],
        "description": "Bunong Wordlist is based on the Bible wordlist of 1,157 unique entries with their frequencies.",
        "id": "sil.cmo.bw",
        "name": "Bunong Wordlist",
        "authorName": "[object Object]",
        "lastModifiedDate": "2022-02-18T18:41:02.203Z",
        "packageFilename": "https://keyman.com/go/package/download/model/sil.cmo.bw?version=1.1&update=1",
        "packageFileSize": 107384,
        "jsFilename": "https://downloads.keyman.com/models/sil.cmo.bw/1.1/sil.cmo.bw.model.js",
        "jsFileSize": 286398,
        "packageIncludes": [],
        "version": "1.1",
        "minKeymanVersion": "12.0",
        "sourcePath": "release/sil/sil.cmo.bw"
    }
]
```